addons/crm/models/crm_team.py

```diff
@@ -228,3 +228,11 @@ def _graph_title_and_key(self):
         if self.dashboard_graph_model == 'crm.lead':
             return ['', _('Pipeline: Expected Revenue')] # no more title
         return super(Team, self)._graph_title_and_key()
+
+    @api.multi
+    def unlink(self):
+        ref = self.env.ref("sales_team.team_sales_department", False)
+        if ref and ref in self:
+            error = _("The sale team %s is mandatory thus cannot be removed.")
+            raise UserError(error % ref.display_name)
```

Comment:
This potentially breaks the uninstallation of `crm`, you should instead do something like

```python3
team = self.env.ref(...)
if team and team in self:
    if self.context.get('_force_unlink'):    # this tells us whether we're currently in an uninstall or not
        self -= team                         # ignore record in case of module uninstallation
    else:
        raise UserError(...)
super().unlink()
```

This guarantees that during an uninstall, the problematic record won't be deleted but the other records will